src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -39,16 +39,20 @@ final class JsonCrawler implements JsonCrawlerInterface
     ];
 
     /**
-     * @param resource|string $raw
+     * @param resource|string|array $data
      */
-    public function __construct(
-        private readonly mixed $raw,
-    ) {
-        if (!\is_string($raw) && !\is_resource($raw)) {
-            throw new InvalidArgumentException(\sprintf('Expected string or resource, got "%s".', get_debug_type($raw)));
+    public function __construct(private mixed $data = [])
+    {
+        if (!\is_string($data) && !\is_resource($data) && !\is_array($data)) {
+            throw new \TypeError(\sprintf('Argument #1 ($data) must be of type string, array or resource, %s given.', get_debug_type($data)));
         }
     }
 
+    public function fromJson(mixed $data): CrawlerInterface
```

Comment:
```suggestion
    /*
     * @param resource|string|array $data
     */
    public function fromJson(mixed $data): CrawlerInterface
```

Reply:
The PHP doc is already present in the interface, should we still add it in the implementing class?

Reply:
variance means this could really be widened by the implementation

Reply:
Oh right, `mixed` is a bit special indeed. Comment added